genbindings.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+# Generate (and reasonably test) C bindings
+
+set -e
+# First build the latest c-bindings-gen binary
+cd c-bindings-gen && cargo build && cd ..
+
+# Then wipe all the existing C bindings (because we're being run in the right directory)
+# note that we keep the few manually-generated files first:
+mv lightning-c-bindings/src/c_types/mod.rs ./
+mv lightning-c-bindings/src/bitcoin ./
+
+rm -rf lightning-c-bindings/src
+
+mkdir -p lightning-c-bindings/src/c_types/
+mv ./mod.rs lightning-c-bindings/src/c_types/
+mv ./bitcoin lightning-c-bindings/src/
+
+# Finally, run the c-bindings-gen binary, building fresh bindings.
+SRC="$(pwd)/lightning/src"
+OUT="$(pwd)/lightning-c-bindings/src"
+OUT_TEMPL="$(pwd)/lightning-c-bindings/src/c_types/derived.rs"
+OUT_F="$(pwd)/lightning-c-bindings/include/rust_types.h"
+OUT_CPP="$(pwd)/lightning-c-bindings/include/lightningpp.hpp"
+RUST_BACKTRACE=1 ./c-bindings-gen/target/debug/c-bindings-gen $SRC/ $OUT/ lightning $OUT_TEMPL $OUT_F $OUT_CPP
+
+# Now cd to lightning-c-bindings, build the generated bindings, and call cbindgen to build a C header file
+PATH="$PATH:~/.cargo/bin"
```

Comment:
I think it might make sense to have a Makefile that would allow splitting the different steps, the most rudimentary of which could be make build and make test.

Reply:
I'm open to what you think the split should be, indeed, `genbindings.sh` right now is more of a test script for CI than a "generator" script.